json_rpc/private/jrpc_conv.nim

```diff
@@ -0,0 +1,94 @@
+# json-rpc
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import
+  std/[tables, macros],
+  results,
+  json_serialization,
+  json_serialization/stew/results as jser_results,
+  json_serialization/std/[options, sets, tables]
+
+export
+  options,
+  sets,
+  tables,
+  json_serialization,
+  jser_results
+
+type
+  StringOfJson* = JsonString
+
+createJsonFlavor JrpcConv,
+  automaticObjectSerialization = true,
+  requireAllFields = false
+
+# Avoid templates duplicating the string in the executable.
+const errDeserializePrefix = "Error deserializing stream for type '"
+
+template wrapErrors(reader, value, actions: untyped): untyped =
+  ## Convert read errors to `UnexpectedValue` for the purpose of marshalling.
+  try:
+    actions
+  except SerializationError as err:
+    reader.raiseUnexpectedValue(errDeserializePrefix & $type(value) & "': " & err.msg)
+
+# Bytes.
+
+proc readValue*(r: var JsonReader[JrpcConv], value: var byte) =
+  ## Implement separate read serialization for `byte` to avoid
+  ## 'can raise Exception' for `readValue(value, uint8)`.
+  value = json_serialization.parseInt(r, uint8).byte
+
+proc writeValue*(w: var JsonWriter[JrpcConv], value: byte) =
+  json_serialization.writeValue(w, uint8(value))
+
+# Enums.
+
+proc readValue*(r: var JsonReader[JrpcConv], value: var (enum)) =
+  wrapErrors r, value:
+    value = type(value) json_serialization.readValue(r, uint64)
```

Comment:
this looks like it will defect on non-members of the enum (or worse)

Reply:
removed